### PR TITLE
Migrate annotation editor to react

### LINKED ISSUE #28: Opensearch: Replace angular annotation editor with react editor
Remove angular annotation editor and replace it with react annotation editor.

### PR DESCRIPTION
**Pull request by svennergr**
What this PR does / why we need it:

This PR migrates the Opensearch annotation editor from angular to react.

I did not use `EditorRow`, etc. from `@grafana/experimental` but instead introduced a little bit of styling. Using `@grafana/experimental` gave me some errors coming from a potential issue with having multiple versions of the `emotion` dependency.

Which issue(s) this PR fixes:

Fixes #28

### CODE REVIEW

File: src/components/QueryEditor/AnnotationQueryEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+export function OpenSearchAnnotationsQueryEditor(props: Props) {
+  const annotation = props.annotation!;
+  const onAnnotationChange = props.onAnnotationChange!;
+
+  return (
+    <>
+      <div className="gf-form-group">
+        <div className={styles.queryWrapper}>
+          <QueryField
+            query={annotation.target.query}
```
Comment (joey-grafana) [inline src/components/QueryEditor/AnnotationQueryEditor.tsx:34]:
```suggestion
            query={annotation.target?.query}
```
or we might get something like this.. :D 

Screenshot 2022-06-09 at 17 49 21

**Review by joey-grafana — CHANGES_REQUESTED** [review]
Opening the annotation editor shows the following error:

Also, please note that the default editor appears to be displayed below the new `AnnotationQueryEditor`. We only want the editor you created as it matches the old editor :) 

Screenshot 2022-06-10 at 10 48 13

**Comment by svennergr** [comment]
> Also, please note that the default editor appears to be displayed below the new AnnotationQueryEditor. We only want the editor you created as it matches the old editor :)

Oh yeah, I do had to change https://github.com/grafana/grafana/blob/cb96f51c733460f37d895bbcda3322b5965640bf/public/app/features/annotations/standardAnnotationSupport.ts#L237 in order for this to show as the only editor. 

I will go ahead and open a PR for this in the grafana repository. This should also resolve the errors you are getting.

**Review by joey-grafana — APPROVED** [review]
Looks great, good job buddy :) 

Could you open up the grafana/grafana PR in `standardAnnotationSupport.ts` before you merge this please.

**Comment by svennergr** [comment]
> Could you open up the grafana/grafana PR in `standardAnnotationSupport.ts` before you merge this please.

Corresponding PR in grafana: https://github.com/grafana/grafana/pull/50655

**Comment by joey-grafana** [comment]
As discussed, we won't be merging this pr until we have updated our @ Grafana 7.5.7 packages.

**Pull request merged**